src/averageAges.js

```diff
@@ -56,6 +73,26 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const searchMothers = onlyWithSon
+    ? people.filter(person => person.sex === 'f'
+      && people.some(child => child.mother === person.name
+      && child.sex === 'm'))
+    : people.filter(person => person.sex === 'f'
+      && people.some(child => child.mother === person.name));
+
+  const searchChildren = onlyWithSon
+    ? people.filter((child) =>
+      searchMothers.some((mother) =>
+        child.mother === mother.name
+        && child.sex === 'm'))
+    : people.filter((child) =>
+      searchMothers.some((mother) =>
+        child.mother === mother.name));
+
+  const becomingMotherAge = searchChildren.map(child => child.born
+    - searchMothers.find(person => person.name === child.mother).born);
+
```

Comment:
2. [CODE STYLE] - If you move your shortened return statement in arrow function to a newline, use round brackets

BAD EXAMPLE:
```
const humans = people.map((a, b) =>
    a + b);

const babies = people.map(person => children 
 ? a
 : b
 
```

GOOD EXAMPLE: 
```
const humans = people.reduce((a, b) => (
  a + b
), 0);